alternatives.md

```diff
@@ -0,0 +1,113 @@
+
+Both Babelfish and Ctags provide positional information for identifiers,
+classes, directives, etc.
+
+The main difference is that Ctags does not provide any form of AST, while
+Babelfish provides native language AST as well as language-independent UAST.
+
+Also, Babelfish ecosystem allows making more complex queries over AST.
+
+## [ANTLR](https://github.com/antlr/antlr4)
+
+> ANTLR (ANother Tool for Language Recognition) is a powerful parser generator for reading, processing, executing, or translating structured text or binary files. It's widely used to build languages, tools, and frameworks. From a grammar, ANTLR generates a parser that can build parse trees and also generates a listener interface (or visitor) that makes it easy to respond to the recognition of phrases of interest.
+
+ANTLR is a language generator with support for most existing languages
+and data formats. But parsers usually produce parse trees that are usually
+different from native language AST, thus requiring additional processing
+to be used for analysis. The structure of parse tree is also language-dependent.
+
+Babelfish provides a correct native language AST as well as
+language-independent UAST for all supported languages.
+
+## [Tree-sitter](https://github.com/tree-sitter/tree-sitter)
+
+> Tree-sitter is a C library for incremental parsing, intended to be used via bindings to higher-level languages. It can be used to build a concrete syntax tree for a program and efficiently update the syntax tree as the program is edited. This makes it suitable for use in text-editing programs.
+
+Tree-sitter provides a simplified AST to be able to execute a limited set
+of queries. It usually cannot provide enough details suitable for static
+analysis. Instead, it focuses on performance and implements real-time
+tree diffing.
+
+Babelfish provides a full native language AST as well as language-independent
+UAST for all supported languages. It is suitable for static analysis since
+it preserves all features of original AST.
+
+## [srcML](https://www.srcml.org/)
+
+> The srcML format is an XML representation for source code, where the markup tags identify elements of the abstract syntax for the language. The srcml program is a command line application for the conversion source code to srcML, an interface for the exploration, analysis, and manipulation of source code in this form, and the conversion of srcML back to source code. The current parsing technologies support C/C++, C#, and Java.
+
+srcML defines an XML schema to annotate source code files with AST structure.
+It also provides tools to query and analyze files in this format. An AST
+structure is language-dependent.
+
+Babelfish provides a native language AST with positional information, that
+can be used to generate the same markup. Also, it provides a language-independent
+UAST for all supported languages. It allows performing the same queries
+and analysis for different programming languages.
```

Comment:
Sorry, the distinction \w srcML is not super-clear to me from this summary, except for different approach with XML schema for annotations :/

Reply:
Will try to clarify it. Need some time to check features of each system.

Reply:
Clarified explicit XML-conversion machinery as the biggest difference.